src/user/commands/get-user-info.command.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/database/prisma.service';
+import { UserInfoDTO } from '../dto/user-info.dto';
+
+@Injectable()
+export class GetUserInfo {
+  constructor(private prisma: PrismaService) {}
+
+  async execute(UserId: number): Promise<UserInfoDTO> {
+    const user = await this.prisma.user.findFirst({
+      where: { id: UserId },
+    });
+
+    const student = await this.prisma.student.findFirst({
+      where: { user: { id: user.id } },
+    });
+
+    const userInfo: UserInfoDTO = {
+      name: user.name,
+      email: user.email,
+      description: student?.description || null,
+      enrollment: student?.enrollment || null,
+      course_id: student?.course_id || null,
+      contact_email: student?.contact_email || null,
+      whatsapp: student?.whatsapp || null,
+      linkedin: student?.linkedin || null,
```

Comment:
Se for professor ou coordenador não é pra retornar esses campos, mesmo que eles estejam com `null`. O objeto tem que ter só `name` e `email`.

Reply:
Você também não precisa do `|| null`, o typescript vai colocar `undefined` se não tiver o campo.